File: app/src/main/java/me/wcy/music/receiver/DownloadReceiver.java
package me.wcy.music.receiver;

import android.app.DownloadManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Handler;
import android.text.TextUtils;

import me.wcy.music.R;
import me.wcy.music.application.AppCache;
import me.wcy.music.service.PlayService;
import me.wcy.music.utils.ToastUtils;

/**
 * 下载完成广播接收器
 * Created by hzwangchenyan on 2015/12/30.
 */
public class DownloadReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        long id = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, 0);
        String title = AppCache.getDownloadList().get(id);
        if (!TextUtils.isEmpty(title)) {
            ToastUtils.show(context.getString(R.string.download_success, title));

            // 由于系统扫描音乐是异步执行，因此延迟刷新音乐列表
            Handler handler = new Handler();
            handler.postDelayed(new Runnable() {
                @Override
                public void run() {
                    scanMusic();
                }
            }, 1000);
        }
    }

    private void scanMusic() {
        PlayService playService = AppCache.getPlayService();
        if (playService == null) {
            return;
        }

        playService.updateMusicList(null);
    }
}

File: app/src/main/java/me/wcy/music/receiver/DownloadReceiver.kt
package me.wcy.music.receiver

import android.app.DownloadManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.text.TextUtils
import me.wcy.music.R
import me.wcy.music.application.AppCache
import me.wcy.music.executor.DownloadMusicInfo
import me.wcy.music.utils.ToastUtils
import me.wcy.music.utils.id3.ID3TagUtils
import me.wcy.music.utils.id3.ID3Tags
import java.io.File

/**
 * 下载完成广播接收器
 * Created by hzwangchenyan on 2015/12/30.
 */
class DownloadReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val id = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, 0)
        val downloadMusicInfo: DownloadMusicInfo =
            AppCache.get().getDownloadList().get(id)
        if (downloadMusicInfo != null) {
            ToastUtils.show(context.getString(R.string.download_success, downloadMusicInfo.title))
            val musicPath = downloadMusicInfo.musicPath
            val coverPath = downloadMusicInfo.coverPath
            if (!TextUtils.isEmpty(musicPath) && !TextUtils.isEmpty(coverPath)) {
                // 设置专辑封面
                val musicFile = File(musicPath)
                val coverFile = File(coverPath)
                if (musicFile.exists() && coverFile.exists()) {
                    val id3Tags = ID3Tags.Builder().setCoverFile(coverFile).build()
                    ID3TagUtils.setID3Tags(musicFile, id3Tags, false)
                }
            }
        }
    }
}